packages/core/src/codewhisperer/util/editorContext.ts

```diff
@@ -46,6 +110,44 @@ export function extractContextForCodeWhisperer(editor: vscode.TextEditor): codew
         languageName =
             runtimeLanguageContext.normalizeLanguage(editor.document.languageId) ?? editor.document.languageId
     }
+    if (editor.document.uri.scheme === 'vscode-notebook-cell') {
```

Comment:
can this logic be extracted
```suggestion
    if (editor.document.uri.scheme === 'vscode-notebook-cell') {
        caretLeftFileContext = getNotebookContext(...)
        caretRightFileContext = getNotebookContext(...)
    }
```

Reply:
Pulled out the notebook logic into two functions: one for determining the surrounding notebook, and then one for extending the context if inside a notebook.